bot/exts/evergreen/youtube.py

```diff
@@ -0,0 +1,77 @@
+from typing import Dict, List, Optional
+
+from discord import Embed
+from discord.ext.commands import Bot, Cog, Context, command, cooldown
+from discord.utils import escape_markdown
+
+from bot.constants import Colours, Tokens
+
+KEY = Tokens.youtube
+SEARCH_API = (
+    "https://youtube.googleapis.com/youtube/v3/search?"
+    "part=snippet&type=video&maxResults=5&q={search_term}&key={key}"
+)
+YOUTUBE_URL = "https://www.youtube.com/watch?v={id}"
+RESULT = "`{index}` [{title}]({url}) - {author}"
+
+
+class YouTubeSearch(Cog):
+    """Sends the top 5 results of a query from YouTube."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.http_session = bot.http_session
+
+    async def search_youtube(self, search_term: str) -> Optional[List[Dict[str, str]]]:
+        """Queries API for top 5 results matching the search term."""
+        results = []
+        async with self.http_session.get(
+            SEARCH_API.format(search_term=search_term, key=KEY)
+        ) as response:
+            data = await response.json()
+            for item in data["items"]:
+                results.append(
+                    {
+                        "title": escape_markdown(item["snippet"]["title"]),
+                        "author": escape_markdown(item["snippet"]["channelTitle"]),
+                        "id": item["id"]["videoId"],
+                    }
+                )
+        return results
+
+    @command(name="youtube", aliases=["yt"])
+    @cooldown(1, 15)
```

Comment:
Do we want a global cooldown here? Or would a user bucket be more suitable?

Reply:
I think a user bucket is best, though I'm unsure how to implement it, are there examples elsewhere?

Reply:
This has it, take a look at the whole file though.
https://github.com/python-discord/sir-lancebot/blob/9441c7284e4230b96ce0cd79e6fa86f457530f3c/bot/exts/evergreen/cheatsheet.py#L77